### PR TITLE
fix typos and general language inconsistencies

### PR DESCRIPTION
**Pull request by TessFerrandez**
This PR fixes typos throughout the documentation, general language, broken links, and also some inconsistencies in how we do notes throughout the docs.

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -11,20 +11,24 @@ The Azure TRE management plane consists of two groups of components:
 
 > Shared Services is still work in progress. Please see [#23](https://github.com/microsoft/AzureTRE/issues/23), [#22](https://github.com/microsoft/AzureTRE/issues/21), & [#21](https://github.com/microsoft/AzureTRE/issues/21)
 
-Management API is a service that users can interact with to request changes to workspaces e.g., to create, update, delete workspaces and workspace services inside each workspace. The Composition Service is doing the actual work of mutating the state of each Workspace including the Workspace Services.
+The Management API is a service that users can interact with to request changes to workspaces e.g., to create, update, delete workspaces and workspace services inside each workspace. The Composition Service is doing the actual work of mutating the state of each Workspace including the Workspace Services.
```
Comment (christoferlof) [inline docs/architecture.md:14]:
```suggestion
The TRE API is a service that users can interact with to request changes to workspaces e.g., to create, update, delete workspaces and workspace services inside each workspace. The Composition Service is doing the actual work of mutating the state of each Workspace including the Workspace Services.
```

File: docs/architecture.md [hunk 2]
```diff
@@ -11,20 +11,24 @@ The Azure TRE management plane consists of two groups of components:
...
 
-Shared Services are services available to all Workspaces. Source Mirror can mirror source repositories such as GitHub, but only allowing read-access, hence data from a Workspace cannot be pushed to a source repository.
-Package Mirror is also a read-only front for developer/researcher application package services like NPM, PyPI, and NuGet and operating system application package services like apt-get and Windows Package Manager (winget).
+Shared Services are services available to all Workspaces. **Source Mirror** can mirror source repositories such as GitHub, but only allowing read-access, hence data from a Workspace cannot be pushed to a source repository.
+**Package Mirror** is also a read-only front for developer/researcher application package services like NPM, PyPI, and NuGet and operating system application package services like apt-get and Windows Package Manager (winget).
 
 ## Composition Service
 
 The Composition Service is responsible for managing and mutating Workspaces and Workspace Service.
```
Comment (christoferlof) [inline docs/architecture.md:23]:
```suggestion
The Composition Service is responsible for managing and mutating Workspaces and Workspace Services.
```

File: docs/architecture.md [hunk 3]
```diff
@@ -44,11 +48,11 @@ The Composition Service consists of multiple components.
 | Service Bus | [Azure Service Bus](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-overview) responsible for reliable delivery of messages between components.  |
 | Resource Processor | Responsible for starting the process of mutating a Workspace via a Workspace Template. |
 
-The flow to provision a Workspace is the following (the flow is the same for all kinds of mutations to a Workspace):
+The flow to provision a Workspace is as follows (the flow is the same for all kinds of mutations to a Workspace):
 
-1. A HTTP request to Management API to create a new Workspace. The request contains information like the name of the Workspace, the Workspace Template to use and the parameters required for the Workspace Template (Workspace Templates can via a JSON Schema expose the the parameters).
+1. An HTTP request to Management API to create a new Workspace. The request contains information like the name of the Workspace, the Workspace Template to use, and the parameters required for the Workspace Template (Workspace Templates can expose the parameters via a JSON Schema ).
```
Comment (christoferlof) [inline docs/architecture.md:53]:
```suggestion
1. An HTTP request to the TRE API to create a new Workspace. The request contains information like the name of the Workspace, the Workspace Template to use, and the parameters required for the Workspace Template (Workspace Templates can expose the parameters via a JSON Schema ).
```

**Review by christoferlof — APPROVED** [review]
Considering #724 , I suggest renaming Management API to TRE API already now.

File: docs/architecture.md (hunk 3 above)
Comment (Lybecker) [inline docs/architecture.md]:
```suggestion
1. A HTTP request to Management API to create a new Workspace. The request contains information like the name of the Workspace, the Workspace Template to use, and the parameters required for the Workspace Template (Workspace Templates can expose the parameters via a JSON Schema ).
```

**Comment by Lybecker** [comment]
I have a lot of changes to the documentation in the `defaultroute `branch too. It is going to be fun to merge :)

**Review by Lybecker — APPROVED** [review]
LGTE

File: docs/architecture.md (hunk 3 above)
Comment (TessFerrandez) [reply docs/architecture.md]: I believe HTTP should be an -- check out https://developer.mozilla.org/en-US/docs/Web/HTTP/Messages and https://www.ibm.com/docs/en/cics-ts/5.3?topic=protocol-http-requests

**Comment by TessFerrandez** [comment]
> I have a lot of changes to the documentation in the `defaultroute `branch too. It is going to be fun to merge :)

@lybecker I know - I was hesitating a bit to change those files -- but I am happy to take the merge if you want to do your merge first

**Pull request merged**